Library/Homebrew/os/mac/pathname.rb

```diff
@@ -1,5 +1,9 @@
 require "os/mac/mach"
 
 class Pathname
-  include MachO
+  if ENV["HOMEBREW_RUBY_MACHO"]
+    include RubyMachO
+  else
+    include CctoolsMachO
+  end
```

Comment:
One last nit: I think it's better to put `require "os/mac/ruby_mach"` and `require "os/mac/cctools_mach"` here inside the condition block and remove `Library/Homebrew/os/mac/mach.rb` file. The reason is we should only load one of the libraries instead of both.